mesonbuild/backend/ninjabackend.py

```diff
@@ -2387,6 +2387,73 @@ def get_link_whole_args(self, linker, target):
         target_args = self.build_target_link_arguments(linker, target.link_whole_targets)
         return linker.get_link_whole_for(target_args) if len(target_args) else []
 
+    def guess_library_absolute_path(self, libname, search_dirs, prefixes, suffixes):
+        for directory in search_dirs:
+            for suffix in suffixes:
+                for prefix in prefixes:
+                    trial = os.path.join(directory, prefix + libname + '.' + suffix)
+                    if os.path.isfile(trial):
+                        return trial
+
+    def guess_external_link_dependencies(self, linker, target, commands, internal):
+        # This is a work around for missing dependency file generation in the linker
+        # https://sourceware.org/bugzilla/show_bug.cgi?id=22843
+        search_dirs = []
+        libs = []
+        absolute_libs = []
+
+        build_dir = self.environment.get_build_dir()
+
+        it = iter(commands)
+        try:
+            while True:
+                item = next(it)
+
+                if item in internal:
+                    continue
+
+                if item.startswith('-L'):
+                    if len(item) > 2:
+                        path = item[2:]
+                    else:
+                        path = next(it)
+
+                    if not os.path.isabs(path):
+                        path = os.path.join(build_dir, path)
+
+                    search_dirs.append(path)
+
+                if item.startswith('-l'):
+                    if len(item) > 2:
+                        libs.append(item[2:])
+                    else:
+                        libs.append(next(it))
+
+                if os.path.isabs(item) and self.environment.is_library(item) and os.path.isfile(item):
+                    absolute_libs.append(item)
+
+        except StopIteration:
+            pass
+
+        guessed_dependencies = []
+
+        if hasattr(linker, 'get_library_naming'):
```

Comment:
Which linkers don't implement this? Should we perhaps implement this for those instead?

Reply:
`linker` is actually a compiler iirc. And i think only the C family compilers (c, cpp plus their objective variants) have this.

If we want to change that i would prefer to discuss that in a separate ticket.

Reply:
Also please add a TODO comment about this, with a list of compilers that are affected.

Reply:
I don't think there is much work that short-circuiting could avoid. We need to parse the command line to get the absolute paths. And everything else is inside this if.

Reply:
All right.